### ISSUE TITLE
Codes for Uruguay projects and project hierarchies

### ISSUE THREAD

**Issue by tlacoyodefrijol** [openspending/openspending#1443]
The TL;DR is if we can make a field belong to more than one hierarchy to make crossed classifications. Lorena suggested looking at the Ramo-Programa structure in the pipeline for Mexico. 

This is based on the Uruguay BOOST data pipeline. From Gabriela:The projects in Uruguay are defined by a project code associated to a program [programa] (FUNCIONAL2) y and a section [inciso] (INSTITUCIONAL1) which build the hierarchies to visualize data. (See attachment)
[Hierarchies and measures with spanish labels_20190320.docx](https://github.com/openspending/openspending/files/3071469/Hierarchies.and.measures.with.spanish.labels_20190320.docx)

In the past, the idea was that to build this hierarchy, they needed to have the 2 fields (inciso and programa) twice. In the pipeline we have Programa twice with 2 different labels, but incisco once. The question is: If a field can belong to 2 hierarchies, we would need a new field for the Inciso, to build the entire hierarchy of a project. But if we can use the same field in many hierarchies, we have an extra field for programa. 

   - header: FUNC2
     title: Fun Programa
    options: {}
    columnType: functional-classification:generic:level2:code

    # Activity
    - header: PROGRAM1
    title: Programa
    options: {}
    columnType: activity:generic:program:code

The suggestion from Lorena is looking at the Ramo-Programa structure in the pipeline for Mexico.

**Comment by LorenaGIFT** [comment]
@tlacoyodefrijol I found in the issues this old one that we raised from Mexico in 2017: https://github.com/openspending/openspending/issues/1315 
It is the same case and apparently we didn't finish solving it.

**Comment by akariv** [comment]
Currently it's not possible to directly give two types to a single field.
However, even without a 'generic' solution, it's quite easy to write pipeline specific code to duplicate a field and give the duplicated field a different name and type (so that it belongs in a separate hierarchy).

**Comment by nightsh** [comment]
@tlacoyodefrijol @LorenaGIFT 
There was an issue with the Uruguay data, which prevented the pipeline from running properly. I have corrected the problems and hosted the CSVs in the git tree, but flagging them here so they would maybe get fixed at the source and we won't have to hold static assets in the repository.

The problem is with quoted values. Some rows (256 in total, for the entire data 2011-2019) had unescaped quotes in their values, thus breaking the parser. An example from the 2013 CSV:

```
"2013","Ministerio de Ganadería, Agricultura y Pesca","Dirección General de Secretaría","DESARROLLO PRODUCTIVO","Fortalecimiento de la base productiva de bienes y servicios","800",""Fideicomiso Financiero de Tierras"","Endeudamiento interno","FINANCIACIONES INTERNAS","Inversión","Gastos No Clasificados","Otros Gastos No Clasificados","799.0","Otros Gastos No Clasificados","708200","708200","0","7","1","7","320","10","16","I","7","79"
```

In this case, the offending value was `""Fideicomiso Financiero de Tierras""`.

Here's a quick shell one liner, which I used to identify all the offending values variants:

`grep -hoP '(?<=,)""[^"]*""(?=,)' credito_presupuestal_detalle_201* | sort | uniq`

And here's a command to dig for offending rows in the CSV files:

`grep -P '(?<=,)""[^"]*""(?=,)' credito_presupuestal_detalle_201*`

The fix for this is to either escape the quotes in values or remove them. By escaping `"` would become `\"`, making the offending rows look something like this:

```
"2013","Ministerio de Ganadería, Agricultura y Pesca","Dirección General de Secretaría","DESARROLLO PRODUCTIVO","Fortalecimiento de la base productiva de bienes y servicios","800","\"Fideicomiso Financiero de Tierras\"","Endeudamiento interno","FINANCIACIONES INTERNAS","Inversión","Gastos No Clasificados","Otros Gastos No Clasificados","799.0","Otros Gastos No Clasificados","708200","708200","0","7","1","7","320","10","16","I","7","79"
```

Just for the record, I removed the quotes for the current version, but in case they should be there (in the values), they should be escaped.

**Comment by akariv** [comment]
Pipeline seems to be running OK, but I don't know how to check if the result is what was expected...

**Comment by sglavoie** [comment]
WONTFIX/CLEANUP: Closing old issue without recent activity (~1 year) as part of #1478. Data will be migrated and merged into the DataHub.io platform. GIFT Phase IV was closed. Please feel free to re-open if this is still relevant.
